shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java

```diff
@@ -262,6 +264,38 @@ public void unspecifiedAction(int inputClientId) {
         Arrays.asList(inputClientId, "TextInputAction.unspecified"));
   }
 
+  public void performPrivateCommand(int inputClientId, String action, Bundle data) {
+    HashMap<Object, Object> json = new HashMap<>();
+    json.put("action", action);
+    if (data != null) {
+      HashMap<String, Object> dataMap = new HashMap<>();
+      Set<String> keySet = data.keySet();
+      for (String key : keySet) {
+        Object value = data.get(key);
+        if (value instanceof byte[]) {
+          dataMap.put(key, data.getByteArray(key));
```

Comment:
Only the ByteArray type is required for the Crowdsource Gboard project, but supporting additional types to be compatible with the types in Bundle.

Reply:
Thanks for putting in the work to support all of those and test them all.